File: src/HH20Backend/HH20Core/Models/Share.cs
using System;
using Dapper.Contrib.Extensions;

namespace HH20Core.Models
{
    [Table("Share")]
    public class Share
    {
        [ExplicitKey]
        public Guid Id { get; set; }
        public string ShareKey { get; set; }
        public Guid SharingUserId { get; set; }
        public DateTime ValidFromTime { get; set; }
        public DateTime ValidUntilTime { get; set; }
        public string Email { get; set; }
        public bool AllowActivities { get; set; }
        public bool AllowDocuments { get; set; }
        public bool AllowDiseases { get; set; }
        public bool AllowAcuteDiseases { get; set; }
        public bool AllowComments { get; set; }
        public bool AllowCommentsWrite { get; set; }
    }
}

File: src/HH20Mobile/HealthHack/Server/Model/Share.swift
//
//  Share.swift
//  HealthHack
//
//  Created by Anatoli Benke on 07.11.20.
//  Copyright © 2020 Anatoli Benke. All rights reserved.
//

import Foundation


struct Share: Codable {
    
    var validFromTime: String
    var validUntilTime: String
    var email: String
    var allowActivities: Bool
    var allowDocuments: Bool
    var allowComments: Bool
    var allowCommentsWrite: Bool
    var allowDiseases: Bool
    var allowAcuteDiseases: Bool
    var shareUrl: String?
}
